client/src/components/Sub_RelatedItems/RelProdContainer.jsx

```diff
@@ -38,6 +40,37 @@ export default function RelProdContainer({
       .catch((err) => err);
   }, [parentProductID]);
 
+  // =================== HELPERS ===================
+  const handleRightClick = () => {
+    flushSync(() => {
+      if (index < relatedProds.length - 1) {
+        setIndex(index + 1);
+      } else {
+        setIndex(0);
+      }
+    });
+    cardsRef.current.scrollIntoView({
+      behavior: 'smooth',
+      block: 'nearest',
+      inline: 'center',
+    });
+  };
+
```

Comment:
it seems that outfit and relprod share some codes. i wonder if these functions can be put into a separate js file. so you can just import them to avoid rewriting them

Reply:
Totally agree... refactoring things to a cleaner, DRYer state would make me much happier! Maybe I can pick away at it over next couple days...